engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java

```diff
@@ -240,16 +240,10 @@ private ConstrainedField findPropertyMetaData(Field field) {
 		);
 
 		boolean isCascading = field.isAnnotationPresent( Valid.class );
-		UnwrapValidatedValue unwrapValidatedValue = field.getAnnotation( UnwrapValidatedValue.class );
-		UnwrapMode unwrapMode = UnwrapMode.AUTOMATIC;
-		if ( unwrapValidatedValue != null ) {
-			unwrapMode = unwrapValidatedValue.value() ? UnwrapMode.UNWRAP : UnwrapMode.SKIP_UNWRAP;
-		}
-
 		Set<MetaConstraint<?>> typeArgumentsConstraints = findTypeAnnotationConstraintsForMember( field );
-		if ( !typeArgumentsConstraints.isEmpty() && !ReflectionHelper.isIterable( ReflectionHelper.typeOf( field ) ) ) {
-			unwrapMode = UnwrapMode.UNWRAP;
-		}
+
+		boolean typeArgumentAnnotated = !typeArgumentsConstraints.isEmpty();
+		UnwrapMode unwrapMode = unwrapMode( field, typeArgumentAnnotated );
```

Comment:
We are on the same page. Partly this goes under the long deferred "unification" of the validation process. I'll create an issue.